### DISCUSSION TITLE
ant runinstall - BUILD FAILED

### DISCUSSION THREAD

**Discussion by JannikZorn** [Keyfactor/ejbca-ce#267] [Q&A] [answered]
Hey,
I'm new to EJBCA and tried the native installation instead of docker installation. I followed the documentation until I reached the Installation part. I choosed the Install "EJBCA as CA with a Management CA" and tried to execute the "ant runinstall command" but the build failed.
I'm using Ubuntu 20.04, javadoc 11.0.18 and openjdk 16.0.1 
The received the following Errors:

ejbca:init:
     [echo]
     [echo] ------------------- CA Properties ----------------
     [echo] ca.name                : ManagementCA
     [echo] ca.dn                  : CN=ManagementCA,O=EJBCA Sample,C=SE
     [echo] ca.tokentype           : soft
     [echo] ca.keytype             : RSA
     [echo] ca.keyspec             : 2048
     [echo] ca.signaturealgorithm  : SHA256WithRSA
     [echo] ca.validity            : 3650
     [echo] ca.policy              : null
     [echo] ca.tokenproperties     : ${ca.tokenproperties}
     [echo] httpsserver.hostname   : localhost
     [echo] httpsserver.dn         : CN=localhost,O=EJBCA Sample,C=SE
     [echo] httpsserver.tokentype  : JKS
     [echo] superadmin.cn          : SuperAdmin
     [echo] superadmin.dn          : CN=SuperAdmin
     [echo] superadmin.batch       : true
     [echo] appserver.home         : /opt/wildfly-22.0.0.Final/
     [echo]

ejbca:install:

ejbca:initCA:
     [echo] Initializing CA with 'ManagementCA' 'CN=ManagementCA,O=EJBCA Sample,C=SE' 'soft' '<ca.tokenpassword hidden>' '2048' 'RSA' '3650' 'null' 'SHA256WithRSA'   -superadmincn 'SuperAdmin'...
     [java] Exception in thread "main" java.util.ServiceConfigurationError: org.ejbca.ui.cli.infrastructure.command.CliCommandPlugin: Provider org.ejbca.ui.cli.ra.AddEndEntityCommand could not be instantiated
     [java]     at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
     [java]     at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:804)
     [java]     at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:722)
     [java]     at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1393)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.<init>(CommandLibrary.java:58)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.<clinit>(CommandLibrary.java:39)
     [java]     at org.ejbca.ui.cli.EjbcaEjbCli.main(EjbcaEjbCli.java:32)
     [java] Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private static final java.lang.reflect.Method jdk.proxy2.$Proxy5.m0 accessible: module jdk.proxy2 does not "opens jdk.proxy2" to unnamed module @4434095f
     [java]     at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
     [java]     at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
     [java]     at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:177)
     [java]     at java.base/java.lang.reflect.Field.setAccessible(Field.java:171)
     [java]     at org.jboss.ejb.client.EJBProxyInformation$1.doCompute(EJBProxyInformation.java:101)
     [java]     at org.jboss.ejb.client.EJBProxyInformation$1.computeValue(EJBProxyInformation.java:72)
     [java]     at org.jboss.ejb.client.EJBProxyInformation$1.computeValue(EJBProxyInformation.java:66)
     [java]     at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:228)
     [java]     at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:210)
     [java]     at java.base/java.lang.ClassValue.get(ClassValue.java:116)
     [java]     at org.jboss.ejb.client.EJBProxyInformation.forViewType(EJBProxyInformation.java:242)
     [java]     at org.jboss.ejb.client.EJBLocator.getProxyInformation(EJBLocator.java:375)
     [java]     at org.jboss.ejb.client.EJBLocator.getProxyConstructor(EJBLocator.java:370)
     [java]     at org.jboss.ejb.client.EJBLocator.createProxyInstance(EJBLocator.java:387)
     [java]     at org.jboss.ejb.client.EJBClient.createProxy(EJBClient.java:161)
     [java]     at org.jboss.ejb.client.EJBRootContext.lookupNative(EJBRootContext.java:166)
     [java]     at org.wildfly.naming.client.AbstractContext.lookup(AbstractContext.java:84)
     [java]     at org.wildfly.naming.client.WildFlyRootContext.lookup(WildFlyRootContext.java:144)
     [java]     at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
     [java]     at org.cesecore.jndi.JndiHelper.getRemoteSession(JndiHelper.java:75)
     [java]     at org.cesecore.util.EjbRemoteHelper.getRemoteSession(EjbRemoteHelper.java:79)
     [java]     at org.cesecore.util.EjbRemoteHelper.getRemoteSession(EjbRemoteHelper.java:54)
     [java]     at org.ejbca.ui.cli.ra.AddEndEntityCommand.<init>(AddEndEntityCommand.java:97)
     [java]     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
     [java]     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:78)
     [java]     at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
     [java]     at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
     [java]     at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
     [java]     at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:780)
     [java]     ... 5 more

BUILD FAILED
/opt/ejbca-ce-EJBCA_7_11_0_20221202/build.xml:72: The following error occurred while executing this line:
/opt/ejbca-ce-EJBCA_7_11_0_20221202/bin/cli.xml:98: The following error occurred while executing this line:
/opt/ejbca-ce-EJBCA_7_11_0_20221202/bin/cli.xml:126: The following error occurred while executing this line:
/opt/ejbca-ce-EJBCA_7_11_0_20221202/bin/cli.xml:527: Java returned: 1

Total time: 1 minute 1 second

**Comment by primetomas** [comment]
It is likely your use of OpenJDK 16. Use OpenJDK 11 now.
Soon support for OpenJDK 17 will be out, but it's 11 for now.

**Reply by crrazychicken** [reply]
I've had other version of ejbca-ce up and going in the past but and mow stuck on getting this new one going.  I've gone through the install instructions several times and end up back at the same point, below, now in windows but previously ended up here several times in ubuntu 22.04

I am using WildFly 26, OpenJDK 11.0.18, MariaDB 11.0.1.0
WildFly is confirmed up and going as a service with the admin web pages available and java -version yields the correct results, MariaDB's included HeidiSQL conencts to the created blank DB of ejbca

I'm including the full server.log as well, I think it's a datasource issue but am not proficient enough to be sure.

server.log suspect entry
2023-04-10 08:21:07,086 ERROR [org.jboss.as.controller.management-operation] (External Management Request Threads -- 1) WFLYCTL0013: Operation ("test-connection-in-pool") failed - address: ([
    ("subsystem" => "datasources"),
    ("data-source" => "EjbcaDS")
]) - failure description: "WFLYJCA0040: failed to invoke operation: WFLYJCA0047: Connection is not valid"
2023-04-10 08:21:07,091 INFO  [org.jboss.as.controller] (External Management Request Threads -- 1) WFLYCTL0183: Service status report
WFLYCTL0186:   Services which failed to start:      service jboss.persistenceunit."ejbca.ear#ejbca": javax.persistence.PersistenceException: [PersistenceUnit: ejbca] Unable to build Hibernate SessionFactory

[server.log](https://github.com/Keyfactor/ejbca-ce/files/11191668/server.log)


PS C:\ejbca> ant runinstall
ANT_OPTS is set to  -Djava.security.manager=allow
Buildfile: C:\ejbca\build.xml

customejbca.message:
     [echo] No custom changes to merge.
[available] DEPRECATED - <available> used to override an existing property.
[available]   Build file should not reuse the same property name for different values.

appserver.error.message:

runinstall:

check:bootstrapdone:

ejbca:prompt:
    [input] skipping input as property ca.name has already been set.
    [input] skipping input as property ca.dn has already been set.
    [input] skipping input as property ca.keytype has already been set.
    [input] skipping input as property ca.keyspec has already been set.
    [input] skipping input as property ca.signaturealgorithm has already been set.
    [input] skipping input as property ca.validity has already been set.
    [input] skipping input as property ca.policy has already been set.
     [echo] To set httpsserver.hostname you need to configure conf/web.properties. See sample config file conf/web.properties.sample.
    [input] skipping input as property httpsserver.hostname has already been set.
     [echo]
     [echo] Common Name (CN) of httpsserver dn is by default taken from the httpsserver.hostname.
    [input] skipping input as property httpsserver.dn has already been set.
[server.log](https://github.com/Keyfactor/ejbca-ce/files/11191656/server.log)

    [input] skipping input as property superadmin.cn has already been set.
    [input] skipping input as property superadmin.dn has already been set.
    [input] skipping input as property superadmin.batch has already been set.
    [input] skipping input as property superadmin.password has already been set.
    [input] skipping input as property ca.tokenpassword has already been set.

ejbca:deploytrustprompt:
    [input] skipping input as property java.trustpassword has already been set.

ejbca:deployprompt:
    [input] skipping input as property httpsserver.password has already been set.

ejbca:init:
     [echo]
     [echo] ------------------- CA Properties ----------------
     [echo] ca.name                : BlankCA
     [echo] ca.dn                  : CN=blankCA,O=blank Sample,C=US
     [echo] ca.tokentype           : soft
     [echo] ca.keytype             : RSA
     [echo] ca.keyspec             : 2048
     [echo] ca.signaturealgorithm  : SHA256WithRSA
     [echo] ca.validity            : 3650
     [echo] ca.policy              : null
     [echo] ca.tokenproperties     : ${ca.tokenproperties}
     [echo] httpsserver.hostname   : localhost
     [echo] httpsserver.dn         : CN=blankEJBCA,O=blank,C=US
     [echo] httpsserver.tokentype  : JKS
     [echo] superadmin.cn          : SuperAdmin
     [echo] superadmin.dn          : CN=SuperAdmin
     [echo] superadmin.batch       : true
     [echo] appserver.home         : c:/wildfly
     [echo]

ejbca:install:

ejbca:initCA:
     [echo] Initializing CA with 'blankCA' 'CN=blankCA,O=blank Sample,C=US' 'soft' '<ca.tokenpassword hidden>' '2048' 'RSA' '3650' 'null' 'SHA256WithRSA'   -superadmincn 'SuperAdmin'...
     [java] 07:16:10.710 [main] ERROR org.ejbca.ui.cli.infrastructure.library.CommandLibrary - Error: CLI could not contact EJBCA instance. Either your application server is not up and running, EJBCA has not been deployed successfully, or some firewall rule is blocking the CLI from the application server.
     [java]
     [java] Please be aware that most commands will not work without the application server available.
     [java]
     [java] Exception in thread "main" javax.ejb.NoSuchEJBException: EJBCLIENT000079: Unable to discover destination for request for EJB StatelessEJBLocator for "ejbca/cesecore-ejb/GlobalConfigurationSessionBean", view is interface org.cesecore.configuration.GlobalConfigurationSessionRemote, affinity is None
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:620)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.protocol.remote.RemotingEJBClientInterceptor.handleInvocationResult(RemotingEJBClientInterceptor.java:57)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.TransactionPostDiscoveryInterceptor.handleInvocationResult(TransactionPostDiscoveryInterceptor.java:148)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.handleInvocationResult(DiscoveryEJBClientInterceptor.java:130)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.NamingEJBClientInterceptor.handleInvocationResult(NamingEJBClientInterceptor.java:87)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.call(AuthenticationContextEJBClientInterceptor.java:59)
     [java]     at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.handleInvocationResult(AuthenticationContextEJBClientInterceptor.java:52)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.TransactionInterceptor.handleInvocationResult(TransactionInterceptor.java:212)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java]     at org.jboss.ejb.client.EJBClientInvocationContext.awaitResponse(EJBClientInvocationContext.java:1003)
     [java]     at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:182)
     [java]     at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:116)
     [java]     at com.sun.proxy.$Proxy8.getCachedConfiguration(Unknown Source)
     [java]     at org.ejbca.ui.cli.infrastructure.command.PasswordUsingCommandBase.execute(PasswordUsingCommandBase.java:162)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary$Branch.execute(CommandLibrary.java:309)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary$Branch.execute(CommandLibrary.java:320)
     [java]     at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.findAnblankecuteCommandFromParameters(CommandLibrary.java:88)
     [java]     at org.ejbca.ui.cli.EjbcaEjbCli.main(EjbcaEjbCli.java:37)
     [java]     Suppressed: java.net.ConnectException: Connection refused: no further information
     [java]             at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
     [java]             at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
     [java]             at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:327)
     [java]             at org.xnio.nio.WorkerThread.run(WorkerThread.java:591)
     [java]             at ...asynchronous invocation...(Unknown Source)
     [java]             at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:599)
     [java]             at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:565)
     [java]             at org.jboss.remoting3.ConnectionInfo$None.getConnection(ConnectionInfo.java:82)
     [java]             at org.jboss.remoting3.ConnectionInfo.getConnection(ConnectionInfo.java:55)
     [java]             at org.jboss.remoting3.EndpointImpl.doGetConnection(EndpointImpl.java:499)
     [java]             at org.jboss.remoting3.EndpointImpl.getConnectedIdentity(EndpointImpl.java:445)
     [java]             at org.jboss.remoting3.UncloseableEndpoint.getConnectedIdentity(UncloseableEndpoint.java:52)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider.getConnectedIdentityUsingClusterEffective(RemotingEJBDiscoveryProvider.java:342)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider$DiscoveryAttempt.connectAndDiscover(RemotingEJBDiscoveryProvider.java:436)
     [java]             at org.jboss.ejb.protocol.remote.RemotingEJBDiscoveryProvider.discover(RemotingEJBDiscoveryProvider.java:233)
     [java]             at org.jboss.ejb.protocol.remote.RemoteEJBDiscoveryConfigurator.lambda$configure$0(RemoteEJBDiscoveryConfigurator.java:42)
     [java]             at org.wildfly.discovery.impl.AggregateDiscoveryProvider.discover(AggregateDiscoveryProvider.java:58)
     [java]             at org.wildfly.discovery.ConfiguredProvider.discover(ConfiguredProvider.java:47)
     [java]             at org.wildfly.discovery.Discovery.discover(Discovery.java:122)
     [java]             at org.wildfly.discovery.Discovery.discover(Discovery.java:93)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.discover(DiscoveryEJBClientInterceptor.java:346)
     [java]             at org.wildfly.common.context.Contextual.runBiFunction(Contextual.java:224)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.discover(DiscoveryEJBClientInterceptor.java:341)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.doAnyDiscovery(DiscoveryEJBClientInterceptor.java:502)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.executeDiscovery(DiscoveryEJBClientInterceptor.java:426)
     [java]             at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.handleInvocation(DiscoveryEJBClientInterceptor.java:110)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:514)
     [java]             at org.jboss.ejb.client.NamingEJBClientInterceptor.handleInvocation(NamingEJBClientInterceptor.java:69)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:514)
     [java]             at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.lambda$handleInvocation$0(AuthenticationContextEJBClientInterceptor.java:45)
     [java]             at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.call(AuthenticationContextEJBClientInterceptor.java:59)
     [java]             at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.handleInvocation(AuthenticationContextEJBClientInterceptor.java:44)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:514)
     [java]             at org.jboss.ejb.client.TransactionInterceptor.handleInvocation(TransactionInterceptor.java:205)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:514)
     [java]             at org.wildfly.common.context.Contextual.runExConsumer(Contextual.java:203)
     [java]             at org.jboss.ejb.client.EJBClientInvocationContext.sendRequestInitial(EJBClientInvocationContext.java:341)
     [java]             at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationH
[server.log](https://github.com/Keyfactor/ejbca-ce/files/11191670/server.log)
andler.java:178)
     [java]             ... 7 more

BUILD FAILED
C:\ejbca\build.xml:72: The following error occurred while executing this line:
C:\ejbca\bin\cli.xml:98: The following error occurred while executing this line:
C:\ejbca\bin\cli.xml:126: The following error occurred while executing this line:
C:\ejbca\bin\cli.xml:527: Java returned: 1

**Reply by primetomas** [reply, marked as answer]
Your issue with "EjbcaDS" means that you have not configured your external database correctly in WildFly. WildFly is unable to open a connection to the database. That means you have tons of errors after "ant deployear". Some of these errors typically have good messages from the database driver, i.e. something like "unable to connect", "invalid authentication" or similar. 

Make sure your database connection is working.
